### PR TITLE
FIX: Xdawn with shuffled epochs

### PR DESCRIPTION
**Pull request by kingjr**
`sklearn.cross_validation.ShuffleSplit` shuffle the epochs order and broke Xdawn.

cc @kaichogami @alexandrebarachant

### CODE REVIEW

File: mne/preprocessing/xdawn.py [hunk 1]
```diff
@@ -241,6 +241,10 @@ def fit(self, epochs, y=None):
         self : Xdawn instance
             The Xdawn instance.
         """
+
+        # Ensure epochs order
+        epochs = epochs[np.argsort(epochs.events[:, 0])]
```
Comment (agramfort) [inline mne/preprocessing/xdawn.py]: just sort if not sorted already. Otherwise you make a copy for nothing.

something like

np.all(np.diff(epochs.events[:, 0]) >= 0)

**Comment by agramfort** [comment]
@kingjr I think you found your bug :)

**Comment by agramfort** [comment]
thx @kingjr
